vhost-device-console/src/vhu_console.rs

```diff
@@ -497,72 +501,87 @@ impl VhostUserConsoleBackend {
 
     /// Set self's VringWorker.
     pub fn set_vring_worker(&self, vring_worker: Arc<VringEpollHandler<Arc<RwLock<Self>>>>) {
-        let rx_event_fd = self.rx_event.as_raw_fd();
+        let rx_event_fd = self.rx_event.0.as_raw_fd();
         vring_worker
             .register_listener(
                 rx_event_fd,
-                EventSet::IN,
+                EventSet::READABLE,
                 u64::from(QueueEvents::BACKEND_RX_EFD),
             )
             .unwrap();
 
-        let rx_ctrl_event_fd = self.rx_ctrl_event.as_raw_fd();
+        let rx_ctrl_event_fd = self.rx_ctrl_event.0.as_raw_fd();
         vring_worker
             .register_listener(
                 rx_ctrl_event_fd,
-                EventSet::IN,
+                EventSet::READABLE,
                 u64::from(QueueEvents::BACKEND_CTRL_RX_EFD),
             )
             .unwrap();
 
-        let exit_event_fd = self.exit_event.as_raw_fd();
+        let exit_event_fd = self.exit_event.0.as_raw_fd();
         vring_worker
             .register_listener(
                 exit_event_fd,
-                EventSet::IN,
+                EventSet::READABLE,
                 u64::from(QueueEvents::EXIT_EFD),
             )
             .unwrap();
 
-        let epoll_fd = self.epoll_fd.as_raw_fd();
+        let epoll_fd = self.epoll_fd.lock().unwrap().as_raw_fd();
         vring_worker
-            .register_listener(epoll_fd, EventSet::IN, u64::from(QueueEvents::KEY_EFD))
```

Comment:
I feel wired here. It registers an epoll_fd to another epoll_fd. I have no idea why it does.